lib/ansible/modules/windows/win_data_deduplication.ps1

```diff
@@ -0,0 +1,165 @@
+#!powershell
+
+# Copyright: 2019, rnsc(@rnsc) <github@rnsc.be>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt
+
+#Requires -Module Ansible.ModuleUtils.Legacy
+#Requires -Module Ansible.ModuleUtils.SID
+
+$ErrorActionPreference = "Stop"
+
+$params = Parse-Args $args -supports_check_mode $false
+
+$drive_letter = Get-AnsibleParam -obj $params -name 'drive_letter' -type 'str' -failifempty $true
+$enabled = Get-AnsibleParam -obj $params -name 'enabled' -type 'bool' -default $true
+$settings = Get-AnsibleParam -obj $params -name 'settings' -type 'list' -default $null
+$dedup_job = Get-AnsibleParam -obj $params -name 'dedup_job' -type 'list' -default $null
+
+$result = @{
+	changed = $false
+}
+
+# Enable-DedupVolume -Volume <String>
+
+function Set-DataDeduplication($volume) {
```

Comment:
There are a lot of changes with boolean values here. why do you need both `$is_enabled` and `$dedup_info.Enabled`?

Reply:
@ShachafGoldstein I added for idempotence, if it's already enabled and Ansible passes the "enabled" value, we shouldn't try to enable it, maybe my approach is not good